R/mixOmics.R

```diff
@@ -154,9 +174,124 @@
 #' graphical outputs to explore relationships between two omics data sets.
 #' BioData Mining 5:19.
 #' @keywords multivariate hplot dplot
+#' @name mixOmics
 #' @export
 #' @example ./examples/mixOmics-examples.R
-mixOmics <- function(X,
+NULL
+
+#' @rdname mixOmics
+#' @export
+setGeneric("mixOmics", signature = "X", function(X, ...)
+    standardGeneric("mixOmics"))
+
+#' @rdname mixOmics
+#' @export
+setMethod("mixOmics", signature = c(X = "MultiAssayExperiment"), function(
+        X, experiments, assay.type, col.var, MINT = FALSE, ...){
+    # Check MINT
+    if( !(is.logical(MINT) && length(MINT) == 1L) ){
+        stop("'MINT' must be TRUE or FALSE.", call. = FALSE)
+    }
+    # Get arguments from SE
+    args <- internal_arguments_from_MAE(
+        mae = X, experiments = experiments, assay.type = assay.type,
+        col.var = col.var, MINT = MINT, ...)
+    # Run analysis
+    res <- do.call(internal_mixOmics, args)
+    return(res)
+    }
+)
+
+#' @rdname mixOmics
+#' @export
+setMethod("mixOmics", signature = c(X = "SummarizedExperiment"), function(
+        X, assay.type, col.var, ...){
+    # Get arguments from SE
+    args <- internal_arguments_from_SE(
+        se = X, assay.type = assay.type, col.var = col.var, ...)
+    # Run analysis
+    res <- do.call(internal_mixOmics, args)
+    return(res)
+    }
+)
+
+#' @rdname mixOmics
+#' @export
+setMethod("mixOmics", signature = c(X = "matrix"), function(
+        X,
+        Y,
+        indY,
+        study,
+        ncomp,
+        keepX,
+        keepY,
+        design,
+        tau = NULL,
+        mode = c("regression", "canonical", "invariant", "classic"),
+        scale,
+        tol =  1e-06,
+        max.iter = 100,
+        near.zero.var = FALSE){
+    res <- internal_mixOmics(
+        X,
+        Y,
+        indY,
+        study,
+        ncomp,
+        keepX,
+        keepY,
+        design,
+        tau = NULL,
+        mode = c("regression", "canonical", "invariant", "classic"),
+        scale,
+        tol =  1e-06,
+        max.iter = 100,
+        near.zero.var = FALSE
+    )
+    return(res)
+    }
+)
+
+#' @rdname mixOmics
+#' @export
+setMethod("mixOmics", signature = c(X = "list"), function(
+        X,
+        Y,
+        indY,
+        study,
+        ncomp,
+        keepX,
+        keepY,
+        design,
+        tau = NULL,
+        mode = c("regression", "canonical", "invariant", "classic"),
+        scale,
+        tol =  1e-06,
+        max.iter = 100,
+        near.zero.var = FALSE,
+        ...){
+    res <- internal_mixOmics(
+        X,
+        Y,
+        indY,
+        study,
+        ncomp,
+        keepX,
+        keepY,
+        design,
+        tau = NULL,
+        mode = c("regression", "canonical", "invariant", "classic"),
+        scale,
+        tol =  1e-06,
+        max.iter = 100,
+        near.zero.var = FALSE,
+        ...
+    )
+    return(res)
+    }
+)
+
+internal_mixOmics <- function(X,
```

Comment:
The old mixOmics function is converted into generic function. Otherwise I have not touched the existing code